### PR TITLE
Link `replace` pipeline stage to doc

### LINKED ISSUE #3415: Document the "replace" pipeline stage
**Is your feature request related to a problem? Please describe.**
The `replace` pipeline stage is currently [undocumented](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#pipeline_stages).

### PR DESCRIPTION
**Pull request by sasagarw**
**What this PR does / why we need it**:
Added the link of `replace` pipeline stage to doc.

**Which issue(s) this PR fixes**:
Closes #3415 <issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: docs/sources/clients/promtail/configuration.md [hunk 1]
```diff
@@ -705,6 +706,27 @@ tenant:
...
+#### replace
+
+The replace stage is a parsing stage that parses a log line using
+a regular expression and replaces the log line.
+
+```yaml
+replace:
+  # The RE2 regular expression. Each named capture group will be added to extracted.
+  # Each capture group and named capture group will be replaced with the value given in `replace`
+  expression: <string>
```
Comment (kavirajk) [inline docs/sources/clients/promtail/configuration.md:718]: can we add an example for each one? (expression, source and replace). In the same document string you already written.

**Review by kavirajk — APPROVED** [review]
Thanks @sasagarw :) 

LGTM. Just added one minor nitl

@KMiller-Grafana can you take a look as well? Thanks

File: docs/sources/clients/promtail/configuration.md (hunk 1 above)
Comment (sasagarw) [reply docs/sources/clients/promtail/configuration.md:718]: The examples are already written here @kavirajk 
https://grafana.com/docs/loki/latest/clients/promtail/stages/replace/

File: docs/sources/clients/promtail/configuration.md [hunk 2]
```diff
@@ -705,6 +706,27 @@ tenant:
...
 
+#### replace
+
+The replace stage is a parsing stage that parses a log line using
+a regular expression and replaces the log line.
+
+```yaml
+replace:
+  # The RE2 regular expression. Each named capture group will be added to extracted.
+  # Each capture group and named capture group will be replaced with the value given in `replace`
```
Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/configuration.md:717]: All these suggestions are for formatting.  The goals is to eliminate horizontal scroll bars in the published docs.
```suggestion
  # Each capture group and named capture group will be replaced with the value given in
  # `replace`
```

File: docs/sources/clients/promtail/configuration.md [hunk 3]
```diff
@@ -705,6 +706,27 @@ tenant:
...
+  expression: <string>
+
+  # Name from extracted data to parse. If empty, uses the log message.
+  # The replaced value will be assigned back to soure key
+  [source: <string>]
+
+  # Value to which the captured group will be replaced. The captured group or the named captured group will be
+  # replaced with this value and the log line will be replaced with new replaced values. An empty value will
+  # remove the captured group from the log line.
```
Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/configuration.md:726]:
```suggestion
  # Value to which the captured group will be replaced. The captured group or the named
  # captured group will be replaced with this value and the log line will be replaced with
  # new replaced values. An empty value will remove the captured group from the log line.
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

**Review by KMiller-Grafana — APPROVED** [review]
Thanks for the changes.  Docs look good now.

**Pull request merged**